spot-client/src/common/backend/utils.js

```diff
@@ -85,13 +70,66 @@ function fetchWithRetry(fetchOptions, maxRetries = 3) {
     return internalFetchWithRetry();
 }
 
+/**
+ * @typedef {Object} RESTBackendCalendarEvent
+ * @property {boolean} allDay
+ * @property {string} calendarId
+ * @property {string} description
+ * @property {string} end - The end date as formatted with {@link Date.toISOString()}.
+ * @property {string} eventId
+ * @property {string} meetingLink
+ * @property {string} start - The start date as formatted with {@link Date.toISOString()}.
+ * @property {string} summary - The title ?
+ * @property {boolean} updatable
+ */
+/**
+ * Retrieves the list of calendar events.
+ *
+ * @param {string} serviceEndpointUrl - The URL pointing to the REST endpoint which serves
+ * the calendar events.
+ * @param {string} jwtToken - The JWT token required for authentication.
+ * @returns {Promise<Array<RESTBackendCalendarEvent>>}
+ */
+export function fetchCalendarEvents(serviceEndpointUrl, jwtToken) {
+    const requestOptions = {
+        method: 'GET',
+        mode: 'cors'
+    };
+
+    if (jwtToken) {
+        requestOptions.headers = new Headers();
+        requestOptions.headers.append('Authorization', `Bearer ${jwtToken}`);
+        requestOptions.headers.append('Accept', 'application/json');
+    }
+
+    let url = serviceEndpointUrl;
+
+    if (!url.includes('{tzid}')) {
+        return Promise.reject(`Missing {tzid} template in the URL: ${url}`);
```

Comment:
I think early exit error handling should be done at the top if possible. That way the reader doesn't have to follow any other logic that gets thrown away.